File: mascii2/src/main/java/com/kastkode/mascii2/ParseResult.java
package com.kastkode.mascii2;

import java.util.List;

import com.kastkode.mascii2.musicelements.MetaInfo;
import com.kastkode.mascii2.musicelements.Part;

public class ParseResult {
	String msg;
	
	List<MetaInfo.MetaInfoElement<?>> globalMetas;
	List<Part> parts;
	MasciiParseErrorListener errors = new MasciiParseErrorListener();
	MasciiSyntaxEventListener syntaxListener;

//	public ParseResult(List<? extends MetaInfo.MetaInfoElement> metas, List<Part> parts) {
//		this("", metas, parts);
//	};
	
	public ParseResult(MasciiSyntaxEventListener listener) {
		this.syntaxListener = listener;
	};
	
	
	
	public MasciiSyntaxEventListener getSyntaxListener() {
		return syntaxListener;
	}

	public void setSyntaxListener(MasciiSyntaxEventListener syntaxListener) {
		this.syntaxListener = syntaxListener;
	}

	public String getMsg() {
		return msg;
	}
	public void setMsg(String msg) {
		this.msg = msg;
	}
	public List<MetaInfo.MetaInfoElement<?>> getGlobalMetas() {
		return syntaxListener.getGlobalMetaInfoChanges();
	}
//	public void setGlobalMetas(List<? extends MetaInfo.MetaInfoElement> metas) {
//		this.globalMetas = metas;
//	}
	public List<Part> getParts() {
		return syntaxListener.getParts();
	}
//	public void setParts(List<Part> parts) {
//		this.parts = parts;
//	}

	public MasciiParseErrorListener getErrors() {
		return errors;
	}

	public void setErrors(MasciiParseErrorListener errors) {
		this.errors = errors;
	}
	
	
}

File: mascii2-typescript/src/ParseResult.ts
import type { MasciiSyntaxEventListener } from './MasciiSyntaxEventListener';
import type { MasciiParseErrorListener } from './MasciiParseErrorListener';
import type { Part } from './musicelements/Part';
import type { MetaInfoElement } from './musicelements/MetaInfo';

export class ParseResult {
    private syntaxListener: MasciiSyntaxEventListener;
    errors: MasciiParseErrorListener | null = null;

    constructor(listener: MasciiSyntaxEventListener) {
        this.syntaxListener = listener;
    }

    getGlobalMetas(): MetaInfoElement<unknown>[] {
        return this.syntaxListener.getGlobalMetaInfoChanges();
    }

    getParts(): Part[] | null {
        return this.syntaxListener.getParts();
    }

    getSyntaxListener(): MasciiSyntaxEventListener {
        return this.syntaxListener;
    }
}
